pkg/metrics/metric/counter.go

```diff
@@ -70,12 +72,47 @@ func NewCounterVec(opts CounterOpts, labelNames []string) DeletableVec[Counter]
 	}
 }
 
+// NewCounterVecWithLabels creates a new DeletableVec[Counter] based on the provided CounterOpts and
+// partitioned by the given labels.
+// This will also initialize the labels with the provided values so that metrics with known label value
+// ranges can be pre-initialized to zero upon init.
+//
+// This should only be used when all label values are known at init, otherwise use of the
+// metric vector with uninitialized labels will result in warnings.
+//
+// Note: Disabled metrics will not have their label values initialized.
+//
+// For example:
+//
+//	NewCounterVecWithLabels(CounterOpts{
+//		Namespace: "cilium",
+//		Subsystem: "subsystem",
+//		Name:      "cilium_test",
+//		Disabled:  false,
+//	}, Labels{
+//		{Name: "foo", Values: NewValues("0", "1")},
+//		{Name: "bar", Values: NewValues("a", "b")},
+//	})
+//
+// Will initialize the following metrics to:
+//
+//	cilium_subsystem_cilium_test{foo="0", bar="a"} 0
+//	cilium_subsystem_cilium_test{foo="0", bar="b"} 0
+//	cilium_subsystem_cilium_test{foo="1", bar="a"} 0
+//	cilium_subsystem_cilium_test{foo="1", bar="b"} 0
+func NewCounterVecWithLabels(opts CounterOpts, labels Labels) DeletableVec[Counter] {
```

Comment:
Couldn't we return the instance vs the interface?

Reply:
Looking at this more, I'm a bit more apprehensive to make this change since that will make the rest of the package inconsistent.

Historically avoiding returning interfaces has been good practice but in this case we might be making the code harder to understand, since the returned type is a generic interface.

Reply:
> I'd like to understand the initial decision to return the generic interface types 

I modeled the API after the prometheus API itself which does the exact same thing. My goal was to keep things mostly the same, and just append the features we needed (metadata + enable/disable capabilities). This made changing types easier since you can go from `prometheus.Counter` to `metric.Counter` ect.

Reply:
Interestingly, [NewCounterVec](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#NewCounterVec) still returns a struct compared to everything else (including NewCounter, ... etc).

Reply:
Going to switch it to *gaugeVec, *counterVec, since that actually seems consistent with the prometheus library.